core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -310,7 +310,9 @@ function taxonomy_form_vocabulary_submit($form, &$form_state) {
   }
 
   $vocabulary = $form_state['vocabulary'];
-  entity_form_submit_build_entity('taxonomy_vocabulary', $vocabulary, $form, $form_state);
+  foreach ($form_state['values'] as $key => $value) {
+    $vocabulary->$key = $value;
+  }
```

Comment:
Nice thanks. As vocabularies are no longer entities themselves, this seems correct. This might also end up saving things like the form_token though. Could you call `form_state_values_clean($form_state)` first?